src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -1129,9 +1132,10 @@ internal enum ParentFlags
             Empty = 0,
             IgnoreResult = 1 << 1,
             Call = 1 << 2,
-            MemberAccess = 1 << 3,
+            MemberAccess = 1 << 3, // Any Parent Expression is a MemberExpression

```

Comment:
Regarding your comment here, parent flags currently are propagated through nested expressions until changed. That means it is more of an ancestor than a parent.

But we may reset the parent in TryEmit when we go deeper. It was my initial variant. Then I removed the reset and nothing changed. So if you found test where it does not work, we may need return reset back:

    internal static ParentFlags Reset(ParentFlags p) =>
        p & (ParentFlags.IgnoreResult);

Should we preserve `IsInstanceAccess` as well?